src/main/java/com/google/jenkins/plugins/computeengine/CleanLostNodesWork.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jenkins.plugins.computeengine;
+
+import com.google.api.services.compute.model.Instance;
+import hudson.Extension;
+import hudson.model.PeriodicWork;
+import hudson.model.Slave;
+import jenkins.model.Jenkins;
+import org.jenkinsci.Symbol;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static com.google.jenkins.plugins.computeengine.ComputeEngineCloud.CLOUD_ID_LABEL_KEY;
+import static java.util.Collections.emptyList;
+
+/**
+ * Periodically checks if there are no lost nodes in GCP.
+ * If it finds any they are deleted.
+ */
+@Extension
+@Symbol("cleanLostNodesWork")
+public class CleanLostNodesWork extends PeriodicWork {
+    protected final Logger logger = Logger.getLogger(getClass().getName());
+
+    public long getRecurrencePeriod() {
+        return HOUR;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    protected void doRun() {
+        logger.log(Level.FINEST, "Starting clean lost nodes worker");
+        getClouds().forEach(this::cleanCloud);
+    }
+
+    private void cleanCloud(ComputeEngineCloud cloud) {
+        logger.log(Level.FINEST, "Cleaning cloud " + cloud.getCloudName());
+        List<Instance> remoteInstances = findRemoteInstances(cloud);
+        Set<String> localInstances = findLocalInstances(cloud);
+        remoteInstances.forEach(remote -> checkOneInstance(remote, localInstances, cloud));
+    }
+
+    private void checkOneInstance(Instance remote, Set<String> localInstances, ComputeEngineCloud cloud) {
```

Comment:
For better orthogonality I suggest separating the concerns of checking for an orphaned instance and terminating. Consider changing this method to: private boolean isOrphaned(Instance instance, ...) and having the terminating decision live in cleanCloud(). This allows opens up the option of testing isOrphaned() independently.